server/paymentProviders/stripe/webhook.ts

```diff
@@ -564,42 +566,58 @@ async function paymentMethodAttached(event: Stripe.Event) {
   });
 }
 
-async function validateWebhook(request: Request<unknown, Stripe.Event>): Promise<Stripe.Event> {
-  const event = <Stripe.Event>request.body;
-  let stripeAccount = event.account;
-  if (!stripeAccount) {
-    stripeAccount = config.stripe.accountId;
-  }
-
-  const stripeConnectedAccount = await models.ConnectedAccount.findOne({
-    where: {
-      username: stripeAccount,
-      service: Service.STRIPE,
-    },
-  });
-
-  if (!stripeConnectedAccount) {
-    throw new Error(`Stripe connected account not found: ${stripeAccount}`);
-  }
-
-  let webhookSigningSecret =
-    stripeConnectedAccount.data?.webhookSigningSecret || stripeConnectedAccount.data.stripeEndpointSecret;
+/*
+ * Stripe issuing events are currently using account webhooks, meaning
+ * that each host with virtual card enabled has an account webhook setup to this endpoint.
+ * To validate the webhook we first get the virtual card and associated host.
+ * This will be deprecated and unified when these events are migrated to the connect webhook.
+ */
+async function handleIssuingWebhooks(request: Request<unknown, Stripe.Event>) {
+  let event = <Stripe.Event>request.body;
 
-  if (process.env.NODE_ENV === 'development' && !webhookSigningSecret && process.env.STRIPE_WEBHOOK_SIGNING_SECRET) {
-    webhookSigningSecret = process.env.STRIPE_WEBHOOK_SIGNING_SECRET;
+  let virtualCardId;
+  if (event.type.startsWith('issuing_authorization')) {
+    virtualCardId = (<Stripe.Issuing.Authorization>event.data.object).card.id;
+  } else if (event.type.startsWith('issuing_transaction')) {
+    const transaction = <Stripe.Issuing.Transaction>event.data.object;
+    virtualCardId = get(transaction, 'transaction.card.id', transaction?.card);
+  } else if (event.type.startsWith('issuing_card')) {
+    virtualCardId = (<Stripe.Issuing.Card>event.data.object).id;
   }
 
-  if (!webhookSigningSecret) {
-    throw new Error('Stripe Webhook Signin Secret not set for Host');
+  if (!virtualCardId) {
+    throw new Error('virtual card id not set in webhook event');
   }
 
-  const stripe = StripeCustomToken(stripeConnectedAccount.token);
+  const virtualCard = await getVirtualCardForTransaction(virtualCardId);
+  const stripeClient = await virtualcard.getStripeClient(virtualCard.host);
+  const webhookSigningSecret = await virtualcard.getWebhookSigninSecret(virtualCard.host);
 
   try {
-    return stripe.webhooks.constructEvent(request.rawBody, request.headers['stripe-signature'], webhookSigningSecret);
-  } catch (e) {
+    event = stripeClient.webhooks.constructEvent(
+      request.rawBody,
+      request.headers['stripe-signature'],
+      webhookSigningSecret,
+    );
+  } catch {
     throw new Error('Source of event not recognized');
```

Comment:
It is fine to throw here because the person receiving this is probably trying to forge events and won't try it again.